app/src/main/java/com/omelan/cofi/MainActivity.kt

```diff
@@ -101,6 +102,25 @@ class MainActivity : MonetCompatActivity() {
         }
     }
 
+    private fun isAppInstalled(packageName: String): Boolean {
+        return try {
+            packageManager.getPackageInfo(packageName, 0)
+            true
+        } catch (e: Exception) {
+            false
+        }
+    }
+
+    private fun showInstallPrompt() {
+        // Here it shows a install prompt for instant apps
+        val postInstall = Intent(Intent.ACTION_MAIN)
+            .addCategory(Intent.CATEGORY_DEFAULT)
+            .setPackage("com.omelan.cofi")
+
+        InstantApps.showInstallPrompt(this@MainActivity, postInstall, 1000, null)
+    }
```

Comment:
Move it to separate file maybe something named `InstantUtils` and make it only for `instant` build variant
So it should be in `app/src/instant/java/com/omelan/cofi/utils`

Then to use it in files in `main` source set, you would need to add another same file in `app/src/full/java/com/omelan/cofi/utils` with just stubs of those functions so something like this

```kt
    fun isInstantApp() = false

    fun showInstallPrompt() {  }
```
as we don't need this when the app isn't an "instant" app (especially when downloaded from F-droid, where I cannot include any google play related dependencies) 

If you need some more explanation please get familiar with https://developer.android.com/studio/build/build-variants if you still need some help I'll be happy to get you trough this 😉

Reply:
How do I make InstantUtils for only instant build variant?

Reply:
@rozPierog

Reply:
Path to the file defines for which build variant it will be accessible to. eg. 

- app/src/**instant**/java/com/omelan/cofi/utils - will be only for the `instant` build variant
- app/src/**full**/java/com/omelan/cofi/utils - will be only for the `full` build variant
- app/src/**main**/java/com/omelan/cofi/utils - will be accessible from all build variants
@detherminal